tests/functional/geth/test_provider.py

```diff
@@ -40,22 +41,31 @@ def test_uri(geth_provider):
 
 
 @geth_process_test
-def test_default_public_uri(config):
+def test_uri_localhost_not_running_uses_random_default(config):
     cfg = config.get_config("geth").ethereum.mainnet
     assert cfg["uri"] in PUBLIC_CHAIN_META["ethereum"]["mainnet"]["rpc"]
     cfg = config.get_config("geth").ethereum.sepolia
     assert cfg["uri"] in PUBLIC_CHAIN_META["ethereum"]["sepolia"]["rpc"]
 
 
 @geth_process_test
-def test_uri_uses_value_from_config(geth_provider, temp_config):
+def test_uri_when_configured(geth_provider, temp_config, ethereum):
     settings = geth_provider.provider_settings
     geth_provider.provider_settings = {}
     value = "https://value/from/config"
-    config = {"geth": {"ethereum": {"local": {"uri": value}}}}
+    config = {"geth": {"ethereum": {"local": {"uri": value}, "mainnet": {"uri": value}}}}
     try:
         with temp_config(config):
+            # Assert we use the config value.
             assert geth_provider.uri == value
+
+            # Assert provider settings takes precedence.
+            expected = DEFAULT_SETTINGS["uri"]
+            for network in ("local", "mainnet"):
```

Comment:
`mainnet` example here fails without the change
now, no longer weird - all that is changing here is that if the user specifies the value in `--network ethereum:mainnet:http://localhost:8545` it will use that value instead of the random public one.

But if the user does not do that, it will still use the public one for mainnet and the local one for local by default

Reply:
That works for me if it works for you.